ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/recaptcha/v1/recaptcha/clientlibs/site/js/recaptchawidget.js

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+/**
+ * This class is responsible for interacting with the recaptcha widget. It displays Google reCAPTCHA challenge.
+ */
+
+class RecaptchaWidget {
+    #widget = null
+    #model = null // passed by reference
+    #options = null
+    #lang = 'en'
+
+    constructor(view, model, widget) {
+        // initialize the widget and model
+        this.#widget = widget;
+        this.#model = model;
+        this.#widget = document.createElement("div");
+        this.#widget.classList.add("cmp-adaptiveform-recaptcha__widget");
+        this.#lang = view.formContainer.getModel().lang;
+
+        //Always inserting it in body
+        document.body.appendChild(this.#widget);
+        this.#options = Object.assign({}, {
+            "contextPath": ""
+        }, this.#model._jsonModel);
+
+        this.#renderRecaptcha(widget);
+    }
+
+    #renderRecaptcha(element) {
+
+        var self = this;
+        var recaptchaConfigData = this.#options;
+        element.innerHTML = '<div class="g-recaptcha"></div>';
+        var gcontainer = document.getElementsByClassName("g-recaptcha")[0];
+        var widgetId;
+        var url = recaptchaConfigData.properties["fd:captcha"].uri;
+        if(recaptchaConfigData.recaptchaSize == "invisible") {
+            gcontainer.classList.add('g-recaptcha-invisible');
+        }
+         if (document.getElementsByClassName("g-recaptcha-invisible").length > 0) {
+            guideBridge.getFormModel().subscribe( (action)=>{
+                if (action.type === 'preSubmitValidate') {
+                    return new Promise((resolve) => {
+                        this.recaptchaInvisibleValidate(successCallback, expiredCallback)
+                            .then(() => resolve())
+                            .catch(() => resolve());
+                    });
+                }
+            }, 'preSubmitValidate');
+         }
+
+        var successCallback = function(response) {
+            element.setAttribute("af-grecaptcha-response", response);
+            self.util(response);
+        };
+
+        var expiredCallback = function() {
+            element.removeAttribute("af-grecaptcha-response");
+            grecaptcha.reset(widgetId);
+            self.util("");
+        };
+
+        var onloadCallbackInternal = function() {
+            widgetId = grecaptcha.render(
+                gcontainer,
+                gparameters
+            );
+            return widgetId;
+        };
+
+        var gparameters = {
+            'sitekey': recaptchaConfigData.properties["fd:captcha"].siteKey,
+            'size': recaptchaConfigData.size,
```

Comment:
the jsonModel should not have size, theme and type as first class objects unless we support then in captcha field type. pls have them in properties object instead.